### ISSUE TITLE
Make AI SDK compatible with Together AI

### ISSUE THREAD

**Issue by janpawellek** [vercel/ai#1865]
### Feature Description

Together AI is an official partner of Vercel for the integration of AI: https://vercel.com/integrations/together-ai

Unfortunately, the Together AI API cannot currently be used for generating or streaming because Together AI returns `null` for `tool_calls` if no tools are called (even if tools are not defined at all). This generates the following validation error:

```
Error: Type validation failed: Value: {"id":"88f8b3b34d89b397-MUC","object":"chat.completion.chunk","created":1717680049,"choices":[{"index":0,"text":" I","logprobs":null,"finish_reason":null,"delta":{"token_id":315,"role":"assistant","content":" I","tool_calls":null}}],"model":"mistralai/Mixtral-8x7B-Instruct-v0.1","usage":null}.
Error message: [
  {
    "code": "invalid_union",
    "unionErrors": [
      {
        "issues": [
          {
            "code": "invalid_type",
            "expected": "array",
            "received": "null",
            "path": [
              "choices",
              0,
              "delta",
              "tool_calls"
            ],
            "message": "Expected array, received null"
          }
        ],
        "name": "ZodError"
      },
      {
        "issues": [
          {
            "code": "invalid_type",
            "expected": "object",
            "received": "undefined",
            "path": [
              "error"
            ],
            "message": "Required"
          }
        ],
        "name": "ZodError"
      }
    ],
    "path": [],
    "message": "Invalid input"
  }
]
```

The schema already allows `tool_calls` to be optional at this point, it just needs to be made nullable too.

The same issue occurs for `tool_calls[0].function.name` and `tool_calls[0].function.arguments` which are already optional in the schema, but need to allow null too.

### Use Case

Using the AI SDK with Together AI, an official integration partner of Vercel.

**Comment by lgrammel** [comment]
#1866

**Comment by janpawellek** [comment]
Thank you! Function name and arguments ([lines 465-466](https://github.com/vercel/ai/blob/a68fe74ac9771c8c9c7146bec02bbcf7ca1bea1d/packages/openai/src/openai-chat-language-model.ts#L465-L466)) also have to be nullish, since Together AI sends tool-deltas where either one of them is null.

Don't know if I can re-open this issue somewhere?

**Comment by lgrammel** [comment]
#1870

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/releases/tag/%40ai-sdk/openai%400.0.24

**Comment by janpawellek** [comment]
You guys are great, thank you so much!

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
The issue reports concrete validation errors when processing valid Together AI responses through the existing OpenAI-compatible provider. The schema's failure to accept nullable fields is broken compatibility rather than merely a request for new functionality.

### Evidence
- Together AI responses with null `tool_calls` fail OpenAI-compatible response validation because the schema expects an array.
- Together AI also sends null tool function names or arguments, while the schema only permits those fields to be optional.
- Maintainer-linked fixes #1866 and #1870 and the subsequent `@ai-sdk/openai@0.0.24` release addressed the failures.
